src/main/resources/sqlUpdates/DatabaseChangeGuidelines.md

```diff
@@ -154,28 +117,46 @@ E.g.,
+Then append them into the current version's changeset file in *src/main/resources/sqlUpdates/changeLog-xxxx.xml*
 
+*DON'T EDIT EXISTING CHANGESETS!!!!!* Liquibase uses checksums to determine if changesets have altered and complains 
+- it cannot tell if the database is inconsistent or if it's just an altered changeset.
 
-Then copy the contents of the changeset  and append them into the current version's changeset file in *src/main/resources/sqlUpdates/changeLog-xxxx.xml*
+*If you're not sure what changes to include, then please seek advice before committing the changes to Git.*
 
-*DON'T EDIT EXISTING CHANGESETS!!!!!* Liquibase uses checksums to determine if changesets have altered and complains - it cannot tell if the database is inconsistent or if it's just an altered changeset.
+### Auto-generating the changesets by comparing to the schema created out of Hibernate/JPA annotations 
 
-*If you're not sure what changes to include, then please seek advice before committing the changes to Git.*
+In `pom.xml` file there is a profile that can generate database schema out of Hibernate/JPA annotations, 
+using `hibernate4-maven-plugin`. To run the maven task & generate such schema you need to first 
+create a separate database called `rspace_hib`, you can do that with the following mysql commands:
 
+        drop database rspace_hib;
+        CREATE DATABASE rspace_hib collate 'utf8mb4_unicode_ci';
+        GRANT ALL ON rspace_hib.* TO 'rspacedbuser'@'localhost';
 
-Now, run:
- 
-    mvn -e test -Denvironment=keepdbintact -PtestLiquibase
+Then you should start the RSpace with `-Denvironment=drop-recreate-hibernate-db` param (rather than
+standard `-Denvironment=drop-recreate-db`), which will generate the schema out of the annotations,
+then start the webapp but with all data coming from `rspace_hib` database.
+
+Next, you can compare a regular database schema created from baseline script and  liquibase updates 
+(i.e. one in your regular `rspace` database), to the schema auto-generated from Hibernate/JPA 
+annotations (i.e. one in your new `rspace_hib` database).
 
-This will attempt to apply new liquibase changes to the testLiquibaseUpdate and run JUnit tests.
-All the updates are applied via Spring at startup - since all database access in both tests and the application is through Spring beans, this should work fine.
+You can compare manually, or run another maven command:
+
+    mvn -e process-resources -Denvironment=keepdbintact -PgenerateDiff
 
-4. After they pass, it's OK to commit. If the tests fail, read the maven output to see what went wrong, edit the liquibase changesets that didn't run and try again. Don't edit changesets that completed successfully - Liquibase might try to run them again. 
+which is based on `liquibase-maven-plugin` and will compare your `rspace` database with `rspace_hib`
+database, and generate a time-stamped changeset file in folder *generatedDBChangeSets/* in your 
```

Comment:
On linux I had to create `generatedDBChangeSets` before running the command otherwise it failed, which will probably be the case for others. 

Could be worth mentioning here, or just create the file in the root of the repo.